packages/core/src/account/types.ts

```diff
@@ -1,11 +1,18 @@
 import type { Address } from "abitype";
-import type { Hash, Hex } from "viem";
+import type { Hash, Hex, Transport } from "viem";
 import type { SignTypedDataParameters } from "viem/accounts";
+import type { z } from "zod";
+import type { SupportedTransports } from "../client/types";
 import type { SmartAccountSigner } from "../signer/types";
 import type { BatchUserOperationCallData } from "../types";
+import type { BaseSmartAccountParamsSchema } from "./schema";
 
 export type SignTypedDataParams = Omit<SignTypedDataParameters, "privateKey">;
 
+export type BaseSmartAccountParams<
+  TTransport extends SupportedTransports = Transport
+> = z.infer<ReturnType<typeof BaseSmartAccountParamsSchema<TTransport>>>;
```

Comment:
`z.infer<ReturnType` sounds a lot like `z.output`?

Reply:
without the `RetumType` I get: 
<img width="1053" alt="image" src="https://github.com/alchemyplatform/aa-sdk/assets/43521356/9405c8a3-0408-457a-9629-0896ecbe2d96">

I can do `z.output<ReturnType<typeof ...>` but don't think it makes a difference?